expertsystem/topology.py

```diff
@@ -320,9 +320,7 @@ def __init__(
 
     def __repr__(self) -> str:
         return (
-            f"{self.__class__.__name__}()"
-            f"\n    nodes: {self.nodes}"
-            f"\n    edges: {self.edges}"
+            f"{self.__class__.__name__}(nodes={self.nodes}, edges={self.edges})"
             f"\n    node props: {self.node_props}"
             f"\n    node props: {self.edge_props}"
```

Comment:
should be `edge props:`

Reply:
131b22cf
Actually this `__repr__` is still nonsensical and it's also still quite obscure when printing e.g. a `list` of `StateTransitionGraphs` (namely the solutions the STM finds), but a for a better `__repr__` we may have to reconsider the constructor of this class (perhaps even implement a `__new__`).

Reply:
Thats true...